### PR TITLE
OwnerCompanyID integration to SummitOrder model

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=2896766&fileview=grid

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/Models/Foundation/Summit/Registration/SummitOrder.php [hunk 1]
```diff
@@ -455,35 +455,35 @@ public function setOwnerEmail(string $owner_email): void
     /**
      * @return string
      */
-    public function getOwnerCompany(): ?string
+    public function getOwnerCompanyName(): ?string
     {
-        if($this->hasCompany())
-            return $this->company->getName();
-        return $this->owner_company;
+        if($this->hasCompany() && !is_null($this->owner_company))
```
Comment (smarcet) [inline app/Models/Foundation/Summit/Registration/SummitOrder.php:460]: check for null is redundant
and we need to refactor hasCompany to hasCompanyOwner

File: app/Models/Foundation/Summit/Factories/SummitOrderFactory.php [hunk 2]
```diff
@@ -60,7 +61,15 @@ public static function populate(Summit $summit, SummitOrder $order, array $paylo
             $order->setOwnerEmail(trim($payload['owner_email']));
 
         if (isset($payload['owner_company']) && !is_null($payload['owner_company']))
-            $order->setOwnerCompany(trim($payload['owner_company']));
+            $order->setOwnerCompanyName(trim($payload['owner_company']));
+
+        if (isset($payload['owner_company_id']) && !is_null($payload['owner_company_id'])) {
+            $company = $summit->getRegistrationCompanyById(intval($payload['owner_company_id']));
```
Comment (smarcet) [inline app/Models/Foundation/Summit/Factories/SummitOrderFactory.php:67]: please throw a validate exception is the company is not found

File: database/migrations/model/Version20220420184724.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+SQL;
+        $this->addSql($sql);
+    }
+
+    /**
+     * @param Schema $schema
+     */
+    public function down(Schema $schema): void
+    {
+
```
Comment (smarcet) [inline database/migrations/model/Version20220420184724.php:41]: please add the rollback

ALTER TABLE `SummitOrder` RENAME COLUMN OwnerCompanyID TO CompanyID;

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review comments

File: app/Models/Foundation/Summit/Registration/SummitOrder.php (hunk 1 above)
Comment (romanetar) [reply app/Models/Foundation/Summit/Registration/SummitOrder.php]: done

File: app/Models/Foundation/Summit/Factories/SummitOrderFactory.php (hunk 2 above)
Comment (romanetar) [reply app/Models/Foundation/Summit/Factories/SummitOrderFactory.php]: done

File: database/migrations/model/Version20220420184724.php (hunk 3 above)
Comment (romanetar) [reply database/migrations/model/Version20220420184724.php]: done

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**